### DISCUSSION TITLE
Trouble with understanding the observability api set ups

### DISCUSSION THREAD

**Discussion by cui086** [jonatan-ivanov/teahouse#9] [Q&A] [answered]
Hey Jonatan,

I am trying to migrate my project to spring 3 observability api. I saw this repo from the Youtube video and I found you tracing and loki set up were super helpful. 

I learned observability from  https://spring.io/blog/2022/10/12/observability-with-spring-boot-3 but realized your set up were a bit different. 

What I understand:
- Prometheus, Loki appender and tempo configurations
- Customized prometheus metrics export

What I do not understand:
- I could not exactly found the place you use the observability api or @Observed annotation?
- How did you set up span attritbues?

Big thanks for the Spring Boot 3 observability guide and the YouTube video, plus the GitHub repo - they're super helpful, especially since there aren't a ton of resources out there for the Observability API yet. Great stuff!

**Comment by jonatan-ivanov** [comment, marked as answer]
Hi,

Thank you for the issue, I converted it to a discussion. I'm happy you found the project useful.

In these apps I don't create any Observations manually because basically everything that I would need is automatically instrumented so the `Observation`s are created by the frameworks/libraries I'm using and I did not need to do anything manually.
I understand that this is not optimal for demoing the usage of the Observation API itself, maybe I should instrument the only business logic these apps has: [building](https://github.com/jonatan-ivanov/teahouse/blob/15357303650023905ec1435a829c7d2e49e1d683/tea-service/src/main/java/org/example/teahouse/tea/service/TeaService.java#L32C1-L47) the tea response from water and tea leaf responses.

If you are curious about the usage of the Observation API, we have docs about it:
- https://micrometer.io/docs/observation
- https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.observability

And also a bunch of samples in a separate repository: https://github.com/micrometer-metrics/micrometer-samples/tree/main/micrometer-samples-boot3-web

For `@Observed` you need to create an `@ObservedAspect` `@Bean` (see docs) on your own but this will not be necessary from Spring Boot 3.2.0: https://github.com/spring-projects/spring-boot/pull/35191

**Comment by cui086** [comment]
Thanks for your info! I will look into those documents you provided!
